calculator.js

```diff
@@ -1,5 +1,164 @@
-const exampleAdditionInput = {
+// calculator.js
+
+// Example Valid Inputs 
+const exampleAdditionInput1 = {
   num1: 3,
   num2: 5,
+  operation: '+',
+}
+
+const exampleAdditionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: 'add',
+}
+
+const exampleSubtractionInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: '-',
+}
+
+const exampleSubtractionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: 'subtract',
+}
+
+const exampleMultiplicationInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: '*',
+}
+
+const exampleMultiplicationInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: 'multiply',
+}
+
+const exampleDivisionInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: '/',
+}
+
+const exampleDivisionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: 'divide',
+}
+
+// Example Invalid Inputs
+const exampleBadInput1 = {
+  num1: 1,
+  num2: 2,
+  operation: 'banana',
+}
+
+const exampleBadInput2 = {
+  num1: [],
+  num2: 'foobar',
+  operation: 'add',
+}
+
+const exampleBadInput3 = {
+  num1: 'foo',
+  num2: 2,
   operation: 'add',
 }
+
+const exampleBadInput4 = {
+  num1: 1,
+  num2: 'bar',
+  operation: 'add',
+}
+
+const exampleBadInput5 = {
+  num1: 1,
+  num2: 0,
+  operation: 'divide',
+}
+
+// Helper Functions 
+const checknum = function(value) {
+  if( typeof value !== 'number' ) {
+    return false;
+  } 
+  return true; 
+} 
+
+const add = (x, y) => x + y ;
+
+const sub =(x, y) => x - y;
+
+const mult = (x, y) => x * y ;
+
+const div = (x, y) => x / y;
+
+// Main Function 
+const calculate = function(user_input) {
+
+  // assign variables 
+  const operator = user_input['operation'] ;
+  const num1 = user_input['num1'];
+  const num2 = user_input['num2']; 
+
+  // check valid nums 
+  if ( (checknum(num1) === false) && (checknum(num2) === false) ) {
+    console.log(`Both \"${num1}\" and \"${num2}\" are invalid numbers.`);
+  } 
+  else if( checknum(num1) === false ) {
+    console.log(`\"${num1}\" is an invalid number.`)
+  } 
+  else if( checknum(num2) === false ) {
+    console.log(`\"${num2}\" is an invalid number.`)
+  }
+
+  // add
```

Comment:
Instead of using `console.log` to output the mathematical equation each time, consider returning the result from the operation (i.e. `add(num1, num2)`) and writing a separate function whose job it is to output the mathematical expression.